### PR TITLE
Never upgrade to an empty guardianSet for Terra

### PR DESCRIPTION
**Pull request by claudijd**
This is largely just footgun avoidance on terra to make sure we never upload an empty GuardianSet, similar how to we check on EVM.

This PR is incomplete as is, and requires that we write some unit-tests for it to ensure the behavior is as expected.

---
This change is [Reviewable](https://reviewable.io/reviews/certusone/wormhole/1033)

### CODE REVIEW

File: terra/contracts/wormhole/src/contract.rs
Comment (kcsongor) [inline terra/contracts/wormhole/src/contract.rs:268]: this error message is a little confusing, because it sounds like we have a guardian whose address is the 0 address, rather than having an empty guardian set. On that note, we should have a check that ensures we don't have a guardian with zero address (at least on EVM)

File: terra/contracts/wormhole/src/contract.rs
Comment (jynnantonix) [inline terra/contracts/wormhole/src/contract.rs:267]: style nit: `if new_guardian_set.addresses.is_empty()`

If you run `cargo clippy` it will catch stuff like this though you might want to apply #983 first to fix unrelated warnings.

File: terra/contracts/wormhole/src/error.rs
Comment (jynnantonix) [inline terra/contracts/wormhole/src/error.rs:72]: There's no harm in making the actual error message more detailed:

```
#[error("Guardian set must not be empty")]
GuardanSetZeroAddresses,
```

File: terra/contracts/wormhole/src/contract.rs
Comment (claudijd) [reply terra/contracts/wormhole/src/contract.rs:267]: I can rebase once we land #983, but I'll also fix this suggestion now.

I like your suggestion on `cargo clippy`, let's just make it part of the build to keep us honest going forward.

**Comment by claudijd** [comment]
Suggestions fixed, please don't merge this until after https://github.com/certusone/wormhole/pull/983 lands.

File: terra/contracts/wormhole/src/error.rs
Comment (claudijd) [reply terra/contracts/wormhole/src/error.rs]: fixed.

File: terra/contracts/wormhole/src/contract.rs
Comment (claudijd) [reply terra/contracts/wormhole/src/contract.rs]: fixed, I left the zero'd address fix out, but I think it's a good idea.  I'll create a ClickUp so I don't miss it, and make sure we get it covered for EVM as well.

File: terra/contracts/wormhole/src/contract.rs
Comment (jynnantonix) [reply terra/contracts/wormhole/src/contract.rs]: This line as also missing a curly brace (`{`) so it won't actually build but maybe that's intentional.

**Review by kcsongor — APPROVED** [review]

File: terra/contracts/wormhole/src/contract.rs
Comment (kcsongor) [inline terra/contracts/wormhole/src/contract.rs:267]: you have a missing `{` here

**Review by kcsongor — CHANGES_REQUESTED** [review]
Other than the syntax error, this looks good to me

**Review by jynnantonix — DISMISSED** [review]
lgtm but please make sure the this PR passes the updated checks in dev.v2

**Review by jynnantonix — APPROVED** [review]

**Pull request closed without merging**
